Existing code updates

cadenkoj opened the pull request:
This is where Koji puts stuff that needs reviewed and merged

Commit:
Update .gitignore

Don't commit lockfiles or the build directory.

Commit:
Update package.json

- Update main to use the build directory's index.js
- Update license to use the correct SPDX license expression
- Change the dev script to build and run nodemon
- Combine the build and start scripts into a prod script which builds and runs
- Install nodemon

Commit:
Delete env.ts

Unnecessary file

Commit:
Update index.ts

Just updates; You can look though them if you want

Commit:
Create utils.ts

Create utils.ts and a function used in index.ts. I don't feel like describing it further, you have eyes

Commit:
Lockfile shit idk

Commit:
Fix untracked files

KojiOdyssey commented:
pls revew

Commit:
Update package.json

- Update description, repository, and remove author. (Add contributors if needed)
- Update scripts

Commit:
Update index.ts

- Minimal updates
- Don't throw error if NODE_ENV is unprovided, default to production

Commit:
Remove packages

lost-in-action commented:
Alright, if @BeakerThe1st is ok we can merge the pull request. There is 1 thing: subcommands need to be made so it doesn't need the parent command code to be modified if you want to add a subcommand. `me`, `uptime`, `ping`, `restart`, `usage` are all in `about.ts`, it would be better to code in a way that it doesn't need `About.ts` to be changed

So if you have subcommands, then the parent must be aware of it: maybe a new member `children` containing a list of subcommands,

cadenkoj commented:
Alrighty

cadenkoj commented:
> why not MIT like the old dot?

I didn't pick the license, BeakerThe1st did

cadenkoj commented:
> ```js
> const tokenENV = "DISCORD_TOKEN",
> devTokenENV = "DISCORD_TOKEN_DEV"
> ```
> 
> Why not add those to the env variables instead of providing them here? It would be cleaner and better to do this and it will be easier to set up

If you mean in .env, they are, there are just some pros to doing this

cadenkoj commented:
Latest comments are because they didn't show up on my phone 😅

BeakerThe1st commented:
> 
> 
> > why not MIT like the old dot?
> 
> I didn't pick the license, BeakerThe1st did

 I don't recall including a license...

BeakerThe1st reviewed (changes requested):
Hi, this file is actually necessary due to the way the dotenv package works with ESM modules, as shown by [this](https://github.com/motdotla/dotenv/issues/133#issuecomment-255298822) GitHub comment!

This is the cleanest implementation, in my view, of the workarounds dotenv provides. https://www.npmjs.com/package/dotenv

cadenkoj commented:
Thank you for all of the feedback @BeakerThe1st, I will work on everything that should be changed and work on improvements soon.

Jasius commented:
> I don't see why there needs to be DISCORD_TOKEN_DEV and DISCORD_TOKEN env vars, ideally the production token would only be present on the production environment

You're right, it's unnecessary. It's just one of the relics from the early implementation as I used to host prod on my local machine from time to time which helped to pinpoint issues better.

Commit:
Keep lockfiles

Commit:
Update SPDX license code

- Switch SPDX license code from GPL-2.0 to MIT
- TODO: Create LICENSE file, I wasn't sure of the name to put, so I held off on doing that

Commit:
Recreate env.ts

Commit:
Update index.ts

- Import env.ts rather than dotenv
- Remove ENV name declarations
- Remove nodeENV variable
- Remove switch statement
- Don't use BOT_TOKEN & BOT_TOKEN_DEV

Commit:
Create yarn.lock

Commit:
Delete utils.ts

Remove utils.ts until needed

BeakerThe1st commented on package.json:
```diff
@@ -1,29 +1,33 @@
+    "start": "node .",
     "build": "tsc",
-    "dev": "node --loader ts-node/esm ./src/index.ts"
+    "prod": "npm run build && node ."
```
Yarn please :)

BeakerThe1st commented on src/index.ts:
```diff
@@ -10,18 +11,25 @@ const dotsimus = new Client({
-  throw new Error("Discord token not provided.");
+
+if (!process.env.NODE_ENV) {
+  process.env.NODE_ENV = 'production';
```
Please don't default to production for NODE_ENV. Either throw an error, or (if you must) default to development.

BeakerThe1st commented on src/index.ts:
```diff
@@ -10,18 +11,25 @@ const dotsimus = new Client({
 
-if (process.env.NODE_ENV === "development") {
-  dotsimus.on("debug", (message) => {
+if (process.env.NODE_ENV == 'development') {
```
Either change it in the prettier config to use single quotes, or continue using double quotes please

Commit:
Update package.json

- Enable importing JSON modules
- Switch from npm to yarn for building

Commit:
Update index.ts

- Don't default to production run environment
- Reimplement the FormatString function
- Use strings.json for neatness
TODO (?) Add more strings to strings.json, example: descriptions (?)

Commit:
Create index.ts

Reimplement the FormatString function

Commit:
Create strings.json

Commit:
Update tsconfig.json

Commit:
Update yarn.lock

BeakerThe1st commented on src/utils/index.ts:
```diff
@@ -0,0 +1,8 @@
+function FormatString(str: string, ... val: string[]):string {
```
Why is this function necessary?

lost-in-action replied:
it's for localization / string mapping stuff I think, like you would write error messages with "{id}" for formatting
could be useful for targeting non-english people, MEE6 has localization support for example

cadenkoj replied:
It might have use in the future, but I agree that it's not necessary at the moment.

BeakerThe1st commented on src/utils/strings.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "UnsuppliedENV": "{0} not supplied. Please confirm `{1}` is set in the .env file!"
```
Unnecessary

BeakerThe1st commented on tsconfig.json:
```diff
@@ -67,6 +67,7 @@
     "skipLibCheck": true /* Skip type checking of declaration files. */,
-    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */
+    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
+    "resolveJsonModule": true
```
Unnecessary

BeakerThe1st commented on src/index.ts:
```diff
@@ -10,20 +13,28 @@ const dotsimus = new Client({
 
-if (!process.env.DISCORD_TOKEN) {
-  throw new Error("Discord token not provided.");
+if (!process.env.NODE_ENV) {
```
Instead of using FormatString, you can either use ternaries in a template literal (preferred) or use a tag function

BeakerThe1st commented on src/index.ts:
```diff
@@ -10,20 +13,28 @@ const dotsimus = new Client({
 });
 
-if (process.env.NODE_ENV === "development") {
+if (process.env.NODE_ENV == "development") {
```
Again, we should be using === instead of == :)

BeakerThe1st reviewed (changes requested):
Looks pretty good, not sure why we need the FormatString function though, also we should generally be using === rather than == in js.

Commit:
Remove unnecessary files

Commit:
Update index.ts
